File: talkadvisor-domain/src/main/java/org/craftsrecords/talkadvisor/recommendation/spi/Profiles.java
package org.craftsrecords.talkadvisor.recommendation.spi;

import org.craftsrecords.talkadvisor.recommendation.profile.Profile;
import org.jetbrains.annotations.NotNull;

public interface Profiles {
    void save(Profile profile);

    @NotNull
    Profile fetch(@NotNull String userId);
}

File: talkadvisor-domain/src/main/kotlin/org/craftsrecords/talkadvisor/recommendation/spi/Profiles.kt
package org.craftsrecords.talkadvisor.recommendation.spi

import org.craftsrecords.hexarch.Repository
import org.craftsrecords.talkadvisor.recommendation.profile.Profile

@Repository
interface Profiles {
    fun save(profile: Profile): Profile

    fun fetch(userId: String): Profile?
}
